### PR TITLE
[MKT-463]: feat/I-been-pawned landing page

### PR DESCRIPTION
**Pull request by jaaaaavier**
Implemeting new Internxt Monitor LP

1.  Created a new LP for a the monitor Tool And their corresponding JSON files for each language, as well as their interface
2. Creation of new components that will be displayed on the landing page when using the tool
3. Adding the tool to the footer to provide access to it
4. Updating components to be able to display the new tool

### CODE REVIEW

File: src/components/monitor/FeatureSection.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  return (
+    <section>
+      <div className="my-10 mb-20 flex flex-col items-center space-y-12 md:my-20 md:mb-32">
+        <div className="mx-auto max-w-full px-4 text-center md:max-w-[774px]">
+          <p className="text-3xl font-semibold text-gray-100 md:text-5xl">{textContent.title}</p>
+          <p className="mt-6 text-lg font-bold text-gray-100 md:mt-12 md:text-xl">{textContent.description}</p>
+          <p className="font-regular mt-6 text-lg text-gray-80 md:mt-12 md:text-xl">{textContent.description2}</p>
+        </div>
+        {/* Cards */}
+        {/* Create Strong Passwords */}
```
Comment (xabg2) [inline src/components/monitor/FeatureSection.tsx]: You can use a loop for that. Also, there is a component that allows you render cards by passing the object.

File: src/components/monitor/InfoSection.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+'use client';
+
+export const InfoSection = ({ textContent }) => {
+  return (
+    <div className="flex flex-col items-center justify-center space-y-10 border-t border-gray-10 pb-10 pt-10 md:flex-row md:space-x-20 md:space-y-0">
+      {/* Pawned Websites */}
+      <div className="flex flex-col items-center justify-center px-5 py-2">
```
Comment (xabg2) [inline src/components/monitor/InfoSection.tsx]: Use a loop

File: src/components/monitor/InfoSection.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        <p className="text-5xl font-semibold text-primary">817</p>
```
Comment (xabg2) [inline src/components/monitor/InfoSection.tsx]: Do not hardcode text. Add it to the JSON instead.

File: src/components/monitor/PwnedElement.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    <div className="flex max-w-[1019px] flex-col pb-5 md:flex-row">
+      <div className="order-1 mb-4 flex items-center justify-center bg-gray-5 p-4 md:order-2 md:mb-0">
+        <div className="flex h-20 w-20 items-center justify-center rounded-full bg-white md:h-24 md:w-24">
+          <p className="text-blue-500 font-bold">Logo</p>
+        </div>
+      </div>
+      <div className="order-2 max-w-full space-y-5 bg-white px-6 pb-6 pt-6 text-center md:order-1 md:max-w-[1019px] md:px-10 md:pb-8 md:pt-8 md:text-left">
+        <p className="text-xl font-bold text-gray-100 md:text-2xl">Canva</p>
+        <p className="text-regular text-base text-gray-80 md:text-lg">
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas faucibus eget enim quis feugiat. Morbi
```
Comment (xabg2) [inline src/components/monitor/PwnedElement.tsx]: Is it correct? Add it to JSON with the desired key so you just have to update the text.

File: src/components/monitor/PwnedElement.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+          condimentum luctus diam, sit amet tincidunt purus.
+        </p>
+        <div className="inline-flex flex-col space-y-1 md:flex-row md:space-x-2 md:space-y-0">
+          <p className="text-base font-bold text-gray-80 md:text-lg">Compromised data:</p>
+          <p className="text-regular text-base text-gray-80 md:text-lg">
+            Email addresses, Geographic locations, Names, Passwords, Usernames
```
Comment (xabg2) [inline src/components/monitor/PwnedElement.tsx]: Add it to the JSON.

File: src/components/monitor/PwnedElement.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        </div>
+      </div>
+      <div className="order-2 max-w-full space-y-5 bg-white px-6 pb-6 pt-6 text-center md:order-1 md:max-w-[1019px] md:px-10 md:pb-8 md:pt-8 md:text-left">
+        <p className="text-xl font-bold text-gray-100 md:text-2xl">Canva</p>
+        <p className="text-regular text-base text-gray-80 md:text-lg">
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas faucibus eget enim quis feugiat. Morbi
+          condimentum luctus diam, sit amet tincidunt purus.
+        </p>
+        <div className="inline-flex flex-col space-y-1 md:flex-row md:space-x-2 md:space-y-0">
+          <p className="text-base font-bold text-gray-80 md:text-lg">Compromised data:</p>
```
Comment (xabg2) [inline src/components/monitor/PwnedElement.tsx]: Add it to the JSON.

File: src/components/monitor/components/EmailToolBar.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+          placeholder={textContent.placeHolder}
+          className="h-14 w-full appearance-none rounded-lg border-2 border-gray-10 bg-white pl-4 pr-14 text-2xl placeholder-gray-30 shadow-subtle outline-none ring-5 ring-primary ring-opacity-0 transition-all delay-150 duration-150 ease-out focus:border-primary focus:ring-opacity-10"
+        />
+        <Button text={'Check'} />
+      </div>
+      <div className="mt-5 flex flex-row items-center space-x-1 text-sm text-gray-50">
+        <Info size={16} data-tooltip-id="email-tooltip" />
+        <span data-tooltip-id="email-tooltip">{textContent.toolTip}</span>
+        <Tooltip id="email-tooltip" place="top">
+          Your email is only checked against databases for breaches.
```
Comment (xabg2) [inline src/components/monitor/components/EmailToolBar.tsx]: Add it to the JSON.

File: src/pages/have-i-been-pwned.tsx [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+import { ComponentsInColumnSection } from '@/components/shared/components/ComponentsInColumnSection';
+import { CardGroup } from '@/components/shared/CardGroup';
+import { CheckSquare, Detective, FacebookLogo, Globe, Info } from '@phosphor-icons/react';
+import SignUpBanner from '@/components/banners/SignUpBanner';
+import { HeroSection } from '@/components/monitor/HeroSection';
+import { InfoSection } from '@/components/monitor/InfoSection';
+import { AllGoodSection } from '@/components/monitor/AllGoodSection';
+import { PwnedSection } from '@/components/monitor/PwnedSection';
+
+const Monitor = ({ lang, metatagsDescriptions, navbarLang, footerLang, langJson, toolsContent, bannerLang }) => {
```
Comment (xabg2) [inline src/pages/have-i-been-pwned.tsx]: Use an interface for that.

File: src/pages/have-i-been-pwned.tsx [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+      description: langJson.FeatureSectionV2.cards![3].description,
+    },
+  ];
+  return (
+    <Layout
+      title={metatags[0].title}
+      description={metatags[0].description}
+      segmentName="Monitor"
+      lang={lang}
+      specialOffer={`https://internxt.com/images/previewLink/LifetimePreviewLink.png`}
```
Comment (xabg2) [inline src/pages/have-i-been-pwned.tsx]: Is this image correct? I don't think so

File: tailwind.config.js [hunk 10]
```diff
@@ -1,4 +1,5 @@
 /* eslint-disable no-dupe-keys */
+const { green } = require('@mui/material/colors');
```
Comment (xabg2) [inline tailwind.config.js]: Why?

File: src/pages/have-i-been-pwned.tsx [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+      <Footer textContent={footerLang} lang={lang} hideNewsletter={false} />
+    </Layout>
+  );
+};
+
+export async function getServerSideProps(ctx) {
+  const lang = ctx.locale;
+
+  const metatagsDescriptions = require(`@/assets/lang/${lang}/metatags-descriptions.json`);
+  const langJson = require(`@/assets/lang/${lang}/i-have-been-pawned.json`);
```
Comment (xabg2) [inline src/pages/have-i-been-pwned.tsx:119]: You have to create a JSON type in `assets/types` to type the Have I been pawned JSON. Then, you can use smth like HaveIBeenPwnedText in the interface.

**Review by xabg2 — CHANGES_REQUESTED** [review]

File: src/pages/have-i-been-pwned.tsx (hunk 8 above)
Comment (jaaaaavier) [reply src/pages/have-i-been-pwned.tsx]: Done!

File: src/pages/have-i-been-pwned.tsx (hunk 11 above)
Comment (jaaaaavier) [reply src/pages/have-i-been-pwned.tsx:119]: Done

File: src/pages/have-i-been-pwned.tsx (hunk 9 above)
Comment (jaaaaavier) [reply src/pages/have-i-been-pwned.tsx]: Removed

File: src/components/monitor/FeatureSection.tsx (hunk 1 above)
Comment (jaaaaavier) [reply src/components/monitor/FeatureSection.tsx]: Done

File: src/components/monitor/InfoSection.tsx (hunk 2 above)
Comment (jaaaaavier) [reply src/components/monitor/InfoSection.tsx]: Done

File: tailwind.config.js (hunk 10 above)
Comment (jaaaaavier) [reply tailwind.config.js]: Removed,I was trying things and I must have forgotten to delete it

File: src/components/monitor/FeatureSection.tsx [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+      <Button
+        text={textContent.cta}
+        className="bg-blue-500 w-full rounded-lg py-4 text-lg font-semibold text-gray-100 md:text-xl"
+        onClick={() => router.push(textContent.redirect)}
+      />
+    </div>
+  );
+};
+
+const FeatureSection = ({ textContent }) => {
```
Comment (xabg2) [inline src/components/monitor/FeatureSection.tsx]: Type the props

File: src/components/monitor/HeroSection.tsx [hunk 13]
```diff
@@ -0,0 +1,22 @@
+'use client';
+import Header from '@/components/shared/Header';
+import EmailToolbar from './components/EmailToolBar';
+
+export const HeroSection = ({ textContent }) => {
```
Comment (xabg2) [inline src/components/monitor/HeroSection.tsx]: Same.

File: src/components/monitor/HeroSection.tsx [hunk 14]
```diff
@@ -0,0 +1,22 @@
+'use client';
```
Comment (xabg2) [inline src/components/monitor/HeroSection.tsx]: Why? We are not using nextJS 14, where all components are SSR by default. Do we need use client for this version?

File: src/components/monitor/InfoSection.tsx [hunk 15]
```diff
@@ -0,0 +1,18 @@
+'use client';
+
+export const InfoSection = ({ textContent }) => {
+  return (
+    <div className="flex flex-col items-center justify-center space-y-10 border-t border-gray-10 pb-10 pt-10 md:flex-row md:space-x-20 md:space-y-0">
+      {[
```
Comment (xabg2) [inline src/components/monitor/InfoSection.tsx]: Extract the array

File: src/components/monitor/InfoSection.tsx [hunk 16]
```diff
@@ -0,0 +1,18 @@
+'use client';
+
+export const InfoSection = ({ textContent }) => {
```
Comment (xabg2) [inline src/components/monitor/InfoSection.tsx]: Type the props

File: src/components/monitor/PwnedElement.tsx [hunk 17]
```diff
@@ -0,0 +1,31 @@
+'use client';
+import { getImage } from '@/lib/getImage';
+import Image from 'next/image';
+
+const PwnedElementCard = ({ textContent }) => {
```
Comment (xabg2) [inline src/components/monitor/PwnedElement.tsx]: Type de props

File: src/components/monitor/PwnedSection.tsx [hunk 18]
```diff
@@ -0,0 +1,33 @@
+'use client';
+
+import { ArrowCircleDown, SmileyMeh, WarningCircle } from '@phosphor-icons/react';
+import PwnedElementCard from './PwnedElement';
+
+export const PwnedSection = ({ textContent }) => {
```
Comment (xabg2) [inline src/components/monitor/PwnedSection.tsx]: Type de props

**Review by xabg2 — CHANGES_REQUESTED** [review]

File: src/components/monitor/PwnedElement.tsx [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+
+export interface PwnedElementCardProps {
+  textContent: HaveIbeenPwnedText['PwnedSection']['pwnedElement'];
+}
+const PwnedElementCard: React.FC<PwnedElementCardProps> = ({ textContent }) => {
+  return (
+    <div className="flex max-w-[1019px] flex-col pb-5 md:flex-row">
+      <div className="order-1 mb-4 flex items-center justify-center bg-gray-5 p-4 md:order-2 md:mb-0">
+        <div className="flex h-20 w-20 items-center justify-center rounded-full bg-white md:h-24 md:w-24">
+          <Image src={getImage(textContent.logoPath)} alt="icon" width={172} height={172} />
```
Comment (xabg2) [inline src/components/monitor/PwnedElement.tsx]: Use better alt here for SEO purposes

**Review by xabg2 — CHANGES_REQUESTED** [review]

File: src/components/monitor/PwnedElement.tsx (hunk 19 above)
Comment (jaaaaavier) [reply src/components/monitor/PwnedElement.tsx]: Done

**Review by xabg2 — APPROVED** [review]

**Pull request merged**